internal/bsr/internal/sign/sign.go

```diff
@@ -29,6 +30,8 @@ type Writer struct {
 	buf *bytes.Buffer
 	w   io.Writer
 	tee io.Writer
+
+	l sync.Mutex
```

Comment:
Could you add a comment to explain what variables this mutex is protecting? Judging by the implementation it seems to be protecting `buf`, `w` _and_ `tee`? It looks like we're taking a single writer input, then creates a tee that writes to both `buf` and `w`, which implies that writes to `tee` would impact `w` and `buf` both, so I think the logic makes sense, but it's unusual to see a single mutex protecting three variables.